### PR TITLE
Improvement ELEMENTS-1178 time date table (Backport to 10.10)

### PR DESCRIPTION
**Pull request by andreacornaglia**

### CODE REVIEW

File: widgets/nuxeo-date.html [hunk 1]
```diff
@@ -22,7 +22,7 @@
 
 <dom-module id="nuxeo-date">
   <template>
-    <span id="datetime" hidden$="[[!datetime]]">[[formatDate(datetime, format)]]</span>
+    <span id="datetime" hidden$="[[!datetime]]">[[formatDateTime(datetime, format)]]</span>
```
Comment (nmpcunha) [inline widgets/nuxeo-date.html:25]: We should not change this since we will break compatibility.

File: package.json [hunk 2]
```diff
@@ -17,7 +17,7 @@
     "eslint": "eslint . --ext html",
     "polylint": "polymer lint",
     "lint": "npm run eslint && npm run polylint",
-    "test": "polymer test --simpleOutput --expanded -l chrome",
+    "test": "polymer test  -l chrome",
```
Comment (nmpcunha) [inline package.json:20]: What is the reason for this change? If you think we need it, I guess it should be done on a separate commit. (And maybe under a different ticket, just to make things organized)

File: test/nuxeo-date-test.html [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+  <test-fixture id="globalconfigTooltipFormat">
+    <template>
+      <nuxeo-date format="LLL"></nuxeo-date>
+    </template>
+  </test-fixture>
+
+  <test-fixture id="differentFormat">
+    <template>
+      <nuxeo-date tformat="relative" tooltipFormat="LLL"></nuxeo-date>
```
Comment (nmpcunha) [inline test/nuxeo-date-test.html:51]: typo `tformat`.

File: test/nuxeo-date-test.html [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+      window.Nuxeo.UI.config[format] = window.Nuxeo.UI.config[format] || value;
+    }
+    /* eslint-disable no-unused-expressions */
+
+    suite('<nuxeo-date>', () => {
+      let element;
+
+      test('Should hide the nuxeo-tooltip if provided format and tooltipFormat are equal', () => {
+        element = fixture('equalFormat');
+        element.date = moment();
```
Comment (nmpcunha) [inline test/nuxeo-date-test.html:70]: If we are defining this after the fixture, I guess we can reuse the same fixture for everything and then set the properties 🤷‍♂️ 
Also, when we use this kind of approach, _flush_ mechanism is usually needed, otherwise we might end not validating what we want.

File: package.json (hunk 2 above)
Comment (andreacornaglia) [reply package.json:20]: not really, I was using this to debug, will revert thanks for the heads-up

File: test/nuxeo-date-test.html (hunk 4 above)
Comment (andreacornaglia) [reply test/nuxeo-date-test.html:70]: yes initially I thought of it but for some reason it wasn't working. Now it is, much more straightforward :)

File: widgets/nuxeo-date.html (hunk 1 above)
Comment (andreacornaglia) [reply widgets/nuxeo-date.html:25]: sorry I copied this from the other version and didn't notice it was different, will revert it.

File: test/nuxeo-tooltip-test.html [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  <script src="../../web-component-tester/browser.js"></script>
+  <script src="../../nuxeo-elements/test/test-helpers.js"></script>
+  <link rel="import" href="../../polymer/polymer.html">
+  <link rel="import" href="../widgets/nuxeo-tooltip.html">
+</head>
+<body>
+
+  <test-fixture id="hidden">
+    <template>
+      <nuxeo-tooltip hidden="true">Hello</nuxeo-tooltip>
```
Comment (nmpcunha) [inline test/nuxeo-tooltip-test.html:37]: Since we are performing a backport, we should keep (as much as possible, of course!) the as it is on master.
```suggestion
      <nuxeo-tooltip hidden>Hello</nuxeo-tooltip>
```

File: test/nuxeo-tooltip-test.html [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    /* eslint-disable no-unused-expressions */
+
+    suite('<nuxeo-tooltip>', () => {
+      let element;
+
+      test('Should not show a paper-tooltip if it has hidden atrribute set to true', () => {
+        element = fixture('hidden');
+
+        element.show();
+        Polymer.dom.flush(element);
```
Comment (nmpcunha) [inline test/nuxeo-tooltip-test.html:58]: I confess I'm a little bit rusty on this stack (little bit different from master) but I believe `Polymer.dom.flush` will be the same as `flush`, as you used on the other test class. Am I correct?

For the same purpose explained above (backport should keep code as much consistent as possible), we should also `await` for the flush.

File: test/nuxeo-tooltip-test.html [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+        const paperTooltip = document.body.querySelector('paper-tooltip');
+        expect(paperTooltip).to.be.null;
+      });
+
+      test('Should show a paper-tooltip with the correct content if no hidden attribute is set', () => {
+        element = fixture('notHidden');
+
+        element.show();
+        Polymer.dom.flush(element);
```
Comment (nmpcunha) [inline test/nuxeo-tooltip-test.html:67]: Same note as above.

File: test/nuxeo-tooltip-test.html [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+        const paperTooltip = document.body.querySelector('paper-tooltip');
+
+        expect(paperTooltip).to.exist;
+        expect(paperTooltip.innerHTML).to.equal('Hello');
```
Comment (nmpcunha) [inline test/nuxeo-tooltip-test.html:71]: It would be interesting to also check the visibility.
We don't have the helper method as on master, but it shouldn't be hard to check. (it may also worth the have it on this repo)

File: test/nuxeo-tooltip-test.html (hunk 6 above)
Comment (andreacornaglia) [reply test/nuxeo-tooltip-test.html:58]: yes you're right! changed.

File: test/nuxeo-tooltip-test.html (hunk 8 above)
Comment (andreacornaglia) [reply test/nuxeo-tooltip-test.html:78]: added it.

File: test/nuxeo-tooltip-test.html [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+
+  <script>
+    /* eslint-disable no-unused-expressions */
+    function isElementVisible(el) {
+      if (!el) {
+        return false;
+      }
+      const styles = window.getComputedStyle(el);
+      return el.offsetWidth !== 0 && el.offsetHeight !== 0 && styles.opacity > 0 && styles.visibility !== 'hidden';
+    }
```
Comment (nmpcunha) [inline test/nuxeo-tooltip-test.html:56]: It may be worth to move it [here](https://github.com/nuxeo/nuxeo-elements/blob/maintenance-2.4.x/test/test-helpers.js), I guess.
Low risk and it would be available for every suite (as on master), since it is generic and useful. 🙂

Comment (andreacornaglia) [reply test/nuxeo-tooltip-test.html:56]: thanks! I wanted to do it but was lost in respect to the repo/branches :)

Comment (andreacornaglia) [reply test/nuxeo-tooltip-test.html:56]: pr here: https://github.com/nuxeo/nuxeo-elements/pull/238
